code/script.js

```diff
@@ -253,91 +257,144 @@ const setSecret = () => {
       )
+      charactersInPlay = charactersInPlay.filter((person) => person[currentQuestion.attribute] === currentQuestion.value);
     } else {
       alert(
-        `No, the person doesn't wear ${attribute}! Remove all that wears ${attribute}`
+      `No, the person doesn't wear ${currentQuestion.attribute}! Remove all that wears ${currentQuestion.attribute}`
       )
+      charactersInPlay = charactersInPlay.filter((person) => person[currentQuestion.attribute] !== currentQuestion.value);
     }
-  } else if (group === 'other') {
-    // Similar to the one above
-  } else {
+  } else if (group === 'other') { //Smokers
+      if (keep) {
+        alert(
+          `Yes, the person is a ${currentQuestion.attribute}! Keep all that are ${currentQuestion.attribute}s`
+        )
+        charactersInPlay = charactersInPlay.filter((person) => person[currentQuestion.attribute] === currentQuestion.value);
+      } else {
+        alert(
+          `No, the person is not a ${currentQuestion.attribute}! Remove all that are ${currentQuestion.attribute}s`
+        )
+        charactersInPlay = charactersInPlay.filter((person) => person[currentQuestion.attribute] !== currentQuestion.value);
+      }
+  } else if (group === 'hair color') {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all persons with yellow hair"
+      alert(
+        `Yes the person has ${attribute} hair! Keep all that have ${attribute} hair!`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person[currentQuestion.attribute] === currentQuestion.value);
     } else {
-      // alert popup that says something like: "NO, the person doesnt have yellow hair! Remove all persons with yellow hair"
+      alert(
+        `No, the person does not have ${attribute} hair! Remove all that have ${attribute} hair`
+       )
+      charactersInPlay = charactersInPlay.filter((person) => person[currentQuestion.attribute] !== currentQuestion.value);
     }
-  }
-
-  // filter to keep or remove based on the keep variable.
-  /* charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-    or 
-    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value) */
-
-  // Invoke a function to redraw the board with the remaining people.
+    
+    } else if (group === 'eye color') {
+      if (keep) {
+        alert (
+          `Yes the person has ${attribute} eyes! Keep all that have ${attribute} eyes!`
+        )
+        charactersInPlay = charactersInPlay.filter((person) => person[currentQuestion.attribute] === currentQuestion.value);
+      } else {
+        alert(
+          `No, the person does not have ${attribute} eyes! Remove all that have ${attribute} eyes`
+        )
+        charactersInPlay = charactersInPlay.filter((person) => person[currentQuestion.attribute] !== currentQuestion.value);
+      }
+    }
+    generateBoard ()
 }
```

Comment:
I offer this comment based on that we chose to solve the problem given in different ways. 

First, I applaud the functionality! 👏

My thought process taking on this problem...
I really wanted to make use of the code provided and got stuck wrapping my head around the filtering and how the global variables were to be implemented throughout the game.

They are all set into play i this section and we know that the global variables are set to: let secret, currentQuestion, charactersInPlay.

- The user makes a simple choice among the questions and this toghether with clicking the find out button  triggers a series of functions.

    1.  We listen for the choice/change with the event listener - selectQuestion
    2. selectQuestion _charges_ the global varible currentQuestion with a value after checking the users catergory choice and validating it against the _secret_
    3. It then goes into the filterCharacters statement and prompting alerts according to selection and then the filter method creates a "copy" of the original board and _charactersInPlay_  on wich it bases the regenerated board.
   
The implied suggestion of use making use of group really throw me off and I just had to change it into something we used earlier in the code, in this case filterCharacters. I don't know if this is really the case but I think this is a reason why I was able to seperate the if statement frpm the filtering and having them execute on different lines instead of on every statement?

This is my code fore reference and I hope this comment makes sense, it atleast helped me clear my own thoughts :)

`const filterCharacters = (keep) => {
  if (currentQuestion.category === 'accessories') {
    if (keep) {
      alert(`Yes, the person wears ${currentQuestion.attribute}! Keep all that wears ${currentQuestion.attribute}`)
    } else {
      alert(`No, the person doesn't wear ${currentQuestion.attribute}! Remove all that wears ${currentQuestion.attribute}`)
    }
  } else if (currentQuestion.category === 'eye color') {
    if (keep) {
      alert(`Yes, the person has ${currentQuestion.value} eyes! Keep all that has ${currentQuestion.value} eyes`)
    } else {
      alert(`No, the person doesn't have ${currentQuestion.value} eyes! Remove all that has ${currentQuestion.value} eyes`)
    }  
  } else if (currentQuestion.category === 'hair color') {
    if (keep) {
      alert(`Yes, the person has ${currentQuestion.value} hair! Keep all that has ${currentQuestion.value} hair`)
    } else {
      alert(`No, the person doesn't have ${currentQuestion.value} hair! Remove all that has ${currentQuestion.value} hair`)
    }  
  } else if (currentQuestion.category === 'other') {
    if (keep) {
      alert(`Yes the person smokes! Keep all the smokers`)
    } else {
      alert(`No the person doesn't smoke! Remove all the smokers`)
    }
  }

  if (keep) {
    charactersInPlay = charactersInPlay.filter((person) => person[currentQuestion.attribute] === currentQuestion.value)
  } else {
    charactersInPlay = charactersInPlay.filter((person) => person[currentQuestion.attribute] !== currentQuestion.value)
  }
  generateBoard()
}`